rust/lance-core/src/utils/tracing.rs

```diff
@@ -76,3 +76,12 @@ pub const IO_TYPE_LOAD_VECTOR_PART: &str = "load_vector_part";
 pub const IO_TYPE_LOAD_SCALAR_PART: &str = "load_scalar_part";
 pub const TRACE_EXECUTION: &str = "lance::execution";
 pub const EXECUTION_PLAN_RUN: &str = "plan_run";
+pub const TRACE_DATASET_EVENTS: &str = "lance::dataset_events";
+pub const CREATE_EVENT: &str = "create";
```

Comment:
Nit: follow the convention of other variable names, should probably be DATASET_CREATE_EVENT (similarly for other ones)